src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/yarn/YarnBomTool.groovy

```diff
@@ -55,41 +80,150 @@ class YarnBomTool extends BomTool implements NestedBomTool<BomToolSearchResult>
     private BomToolSearchResult searchResult;
 
     @Override
-    public BomToolType getBomToolType() {
+    BomToolType getBomToolType() {
         BomToolType.YARN
     }
 
     @Override
-    public boolean isBomToolApplicable() {
-        BomToolSearchResult searchResult = bomToolSearcher.getBomToolSearchResult(sourcePath);
+    boolean isBomToolApplicable() {
+        BomToolSearchResult searchResult = bomToolSearcher.getBomToolSearchResult(sourcePath)
         if (searchResult.isApplicable()) {
-            this.searchResult = searchResult;
-            return true;
+            this.searchResult = searchResult
+            return true
         }
 
-        return false;
+        return false
     }
 
-    public List<DetectCodeLocation> extractDetectCodeLocations(BomToolSearchResult searchResult) {
-        final File yarnLockFile = detectFileManager.findFile(searchResult.searchedDirectory, 'yarn.lock')
-        final List<String> yarnLockText = Files.readAllLines(yarnLockFile.toPath(), StandardCharsets.UTF_8)
-        final DependencyGraph dependencyGraph = yarnPackager.parse(yarnLockText)
-        final ExternalId externalId = externalIdFactory.createPathExternalId(Forge.NPM, searchResult.searchedDirectory.canonicalPath)
-        final def detectCodeLocation = new DetectCodeLocation.Builder(getBomToolType(), searchResult.searchedDirectory.canonicalPath, externalId, dependencyGraph).build()
+    List<DetectCodeLocation> extractDetectCodeLocations(BomToolSearchResult searchResult) {
+        DependencyGraph dependencyGraph
+        def detectCodeLocation
+
+        File yarnLockFile = detectFileManager.findFile(searchResult.searchedDirectory.canonicalPath, 'yarn.lock')
+        yarnLockText = Files.readAllLines(yarnLockFile.toPath(), StandardCharsets.UTF_8)
+        yarnExePath = findExecutablePath(ExecutableType.YARN, true, detectConfiguration.getYarnPath())
+
+        if (detectConfiguration.yarnProductionDependenciesOnly) {
+            dependencyGraph = parseYarnList(readYarnListAsLines(yarnExePath))
+        } else {
+            dependencyGraph = yarnPackager.parseYarnLock(yarnLockText)
+        }
+
```

Comment:
Just above this line is where a split occurs depending on the boolean value of the option that I created above. one route takes a yarn list, the other takes yarn lock, both return a usable DependencyGraph and thus a DetectCodeLocation